packages/engine/lib/memory/src/shared_memory/segment.rs

```diff
@@ -13,6 +13,35 @@ use crate::{
     },
 };
 
+pub struct Buffers<'a> {
```

Comment:
The buffers are still present in the memory, its size may only be set to `0`. This implies, that a buffer is unused but still valid. I didn't use the term "optional" because of this but mentioned, that unneeded buffers may be set to `0`.

Reply:
How is it present in memory if its size is 0? Or is there a concept of capacity vs size here?

Reply:
I think we have a philosophical difference in opinion between the buffer being empty and it not existing.  But to avoid blowing up this PR, am resolving the comment.